src/ui/utils/selection.tsx

```diff
@@ -112,6 +118,23 @@ export const SelectionContextProvider = ({
 
       notify(subscriptions, selectedId);
     };
+    const highlight = (nextHighlightedId) => {
```

Comment:
There should a separate context for highlight functionality. Please, don't mix different things

Reply:
Fixed